src/Hyperopt.jl

```diff
@@ -196,19 +196,30 @@ function pmacrobody(ex, params, ho_, pmap=pmap)
         function workaround_function()
             ho = $(ho_)
             # Getting the history right is tricky when using workers. The approach I've found to work is to
-            # save the actual array (not copy) in hist, temporarily use a new array that will later be discarded
+            # save the actual array (not copy) in hist, temporarily use a channel that will later be discarded
             # reassign the original array and then append the new history. If a new array is used, the change will not be visible in the original hyperoptimizer
             hist = ho.history
             ho.history = []
-            res = $(pmap)(1:ho.iterations) do i
-                $(Expr(:tuple, esc.(params)...)),_ = iterate(ho,i)
+
+            new_history = RemoteChannel(() -> Channel(ho.iterations), 1)
+            $(esc(pmap))(1:ho.iterations) do i
+                $(Expr(:tuple, esc.(params)...)), _ = iterate(ho, i)
                 res = $(esc(ex.args[2])) # ex.args[2] = Body of the For loop
 
-                res, $(Expr(:tuple, esc.(params[2:end])...))
+                put!(new_history, (i, res, $(Expr(:tuple, esc.(params[2:end])...))))
```

Comment:
here, we update `new_history` (which lives on the manager node) from the worker nodes, so that we don't need to care about the result of `pmap`. Why? Because if we pass a custom `on_error`, it would need to return both a loss and the parameters in order to be iterated later, and the user probably doesn't know they can't just pass a loss.

Reply:
Does this comment perhaps deserve to be included in the source as well? I can imagine future me scratching my head over this (Hyperopt.jl has a tendency to require a long warm up before I can do any maintainance on it 🙃)